projects/js-packages/publicize-components/src/components/form/index.js

```diff
@@ -22,41 +24,85 @@ import styles from './styles.module.scss';
  * @param {boolean} props.isPublicizeEnabled            - Whether Publicize is enabled for this post.
  * @param {boolean} props.isRePublicizeFeatureEnabled   - True if the RePublicize feature is available.
  * @param {boolean} props.isPublicizeDisabledBySitePlan - A combination of the republicize feature being enabled and/or the post not being published.
+ * @param {boolean} props.numberOfSharesRemaining       - The number of shares remaining for the current period. Optional.
  * @returns {object}                                    - Publicize form component.
  */
 export default function PublicizeForm( {
 	isPublicizeEnabled,
 	isRePublicizeFeatureEnabled,
 	isPublicizeDisabledBySitePlan,
+	numberOfSharesRemaining = null,
 } ) {
-	const { connections, toggleById, hasConnections } = useSocialMediaConnections();
+	const {
+		connections,
+		toggleById,
+		hasConnections,
+		enabledConnections,
+	} = useSocialMediaConnections();
 	const { message, updateMessage, maxLength } = useSocialMediaMessage();
 
 	const isDisabled = () =>
 		! isRePublicizeFeatureEnabled && connections.every( connection => ! connection.toggleable );
 	const Wrapper = isPublicizeDisabledBySitePlan ? Disabled : Fragment;
 
+	const outOfConnections =
+		numberOfSharesRemaining !== null && numberOfSharesRemaining <= enabledConnections.length;
+
 	return (
 		<Wrapper>
 			{ hasConnections && (
-				<PanelRow>
-					<ul className={ styles[ 'connections-list' ] }>
-						{ connections.map(
-							( { display_name, enabled, id, service_name, toggleable, profile_picture } ) => (
-								<PublicizeConnection
-									disabled={ isRePublicizeFeatureEnabled ? ! isPublicizeEnabled : ! toggleable }
-									enabled={ enabled && ! isPublicizeDisabledBySitePlan }
-									key={ id }
-									id={ id }
-									label={ display_name }
-									name={ service_name }
-									toggleConnection={ toggleById }
-									profilePicture={ profile_picture }
-								/>
-							)
-						) }
-					</ul>
-				</PanelRow>
+				<>
+					{ numberOfSharesRemaining !== null && (
```

Comment:
May be we can do `numberOfSharesRemaining &&  ( ` here?

Reply:
Same as above ^